source/command-logging-and-monitoring/tests/README.rst

```diff
@@ -0,0 +1,53 @@
+.. role:: javascript(code)
+  :language: javascript
+
+==============================
+Command Logging and Monitoring
+==============================
+
+.. contents::
+
+--------
+
+Testing
+=======
+
+Automated Tests
+^^^^^^^^^^^^^^^
+There are tests in the `Unified Test Format <../../unified-test-format/unified-test-format.rst>`__ for both logging and
+monitoring in `/logging/unified </logging/unified>`_ and `/monitoring/unified </monitoring/unified>`_, respectively.
+
+
+Prose Tests
+~~~~~~~~~~~
+Drivers MUST implement the following logging prose tests. These tests require the ability to capture log message data in a
+structured form as described in the 
+`Unified Test Format specification <../../unified-test-format/unified-test-format.rst#expectedLogMessage>`__.
+
+*Test 1: Default truncation limit*
+
+1. Configure logging with a minimum severity level of "debug" for the "command" component. Do not explicitly configure the max document length.
+2. Construct an array ``docs`` containing the document ``{"x" : "y"}`` repeated 100 times.
+3. Insert ``docs`` to a collection via ``insertMany``.
+4. Inspect the resulting "command started" log message and assert that the "command" value is a string of length 1000.
+5. Inspect the resulting "command succeeded" log message and assert that the "reply" value is a string of length <= 1000.
+6. Run ``find()`` on the collection where the document was inserted.
+7. Inspect the resulting "command succeeded" log message and assert that the reply is a string of length 1000.
+
+*Test 2: Explicitly configured truncation limit*
+
+1. Configure logging with a minimum severity level of "debug" for the "command" component. Set the max document length to 5 of (your driver's unit of choice for truncation).
+2. Run the command ``{"hello": true}```.
+3. Inspect the resulting "command started" log message and assert that the "command" value is a string of length 5.
+4. Inspect the resulting "command succeeded" log message and assert that the "reply" value is a string of length 5.
+5. If the driver attaches raw server responses to failures and can access these via log messages to assert on, run the command 
+   ``{"notARealCommand": true}``. Inspect the resulting "command failed" log message and confirm that the server error is
+   a string of length 5.
+
+*Test 3: Truncation with multi-byte codepoints*
```

Comment:
I'm not really sure how drivers can write these tests in a way that is not brittle to the formatting/ordering of commands and replies, as well as the topology you are connected to, since you basically have to find particular byte counts that fall mid code point and set those counts as your max length and see how truncation goes. 

here's the one I came up with for Rust: https://github.com/kmahar/mongo-rust-driver/blob/a03a6990dd8884a93653cd968ac953ca5193dc41/src/test/spec/trace.rs#L137